WordPress/src/main/java/org/wordpress/android/ui/jetpack/JetpackCapabilitiesUseCase.kt

```diff
@@ -96,12 +99,16 @@ class JetpackCapabilitiesUseCase @Inject constructor(
         )
     }
 
-    @Subscribe(threadMode = ThreadMode.BACKGROUND)
+    @Subscribe(threadMode = ThreadMode.MAIN)
     @SuppressWarnings("unused")
     fun onJetpackCapabilitiesFetched(event: OnJetpackCapabilitiesFetched) {
-        dispatcher.unregister(this@JetpackCapabilitiesUseCase)
-        continuation?.resume(event)
-        continuation = null
+        continuation[event.remoteSiteId]?.let {
+            continuation.remove(event.remoteSiteId)
+            it.resume(event)
+        }
+        if (continuation.isEmpty()) {
+            dispatcher.unregister(this@JetpackCapabilitiesUseCase)
```

Comment:
We need to make sure to unregister the instance only when no requests are running (previously, multiple parallel requests for different sites were not supported at all)